internal/models/config.go

```diff
@@ -0,0 +1,171 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package models defines the data models for entities. This file defines the
+// models for V2 configuration format. The defined entities are:
+// ConfigV2 (the top-level structure), Receivers, Exporters, Options, Pipelines.
+package models
+
+/*
+Receivers, Exporters and Options typically have common configuration settings, however
+sometimes specific implementations (e.g. StackDriver) will have extra configuration settings.
+This requires the configuration data for these entities to be polymorphic.
+
+To satisfy these requirements we declare interfaces ReceiverCfg, ExporterCfg, OptionCfg,
+which define the behavior. We also provide helper structs ReceiverCommon, ExporterCommon,
+OptionCommon, which define the common settings and un-marshaling from config files.
+
+Specific Receivers/Exporters/Options are expected to at the minimum implement the
+corresponding interface and if they have additional settings they must also extend
+the corresponding common settings struct (the easiest approach is to embed the common struct).
+
+For example usage see tests.
+*/
+
+// ConfigV2 defines the configuration V2 for the various elements of collector or agent.
+type ConfigV2 struct {
+	Receivers []ReceiverCfg `mapstructure:"-"`
+	Exporters []ExporterCfg `mapstructure:"-"`
+	Options   []OptionCfg   `mapstructure:"-"`
+	Pipelines Pipelines     `mapstructure:"pipelines"`
+}
+
+// ReceiversKeyName is the configuration key name for receivers section.
+const ReceiversKeyName = "receivers"
+
+// ExportersKeyName is the configuration key name for exporters section.
+const ExportersKeyName = "exporters"
+
+// OptionsKeyName is the configuration key name for options section.
+const OptionsKeyName = "options"
+
+// Pipelines defines the pipelines section of configuration.
+type Pipelines struct {
+	Traces  []*Pipeline `mapstructure:"traces"`
+	Metrics []*Pipeline `mapstructure:"metrics"`
+}
+
+// Pipeline defines a single pipeline.
+type Pipeline struct {
+	Name       string              `mapstructure:"name"`
+	Receivers  []PipelineReceiver  `mapstructure:"receivers"`
+	Operations []PipelineOperation `mapstructure:"ordered-operations"`
+	Exporters  []PipelineExporter  `mapstructure:"exporters"`
+}
+
+// PipelineReceiver defines a receiver of a pipeline.
+type PipelineReceiver struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineOperation defines an operation of a pipeline.
+type PipelineOperation struct {
```

Comment:
Are we going with the name Operation instead of Processor? https://github.com/census-instrumentation/opencensus-service/tree/master/processor

Or these are not the same?